src/instrumentation.ts

```diff
@@ -293,66 +293,59 @@ export class Instrumentation extends InstrumentationBase {
+          attributes: Instrumentation.dropInvalidAttributes({
+            [SemanticAttributes.MESSAGING_SYSTEM]:
+              BullMQAttributes.MESSAGING_SYSTEM,
+            [SemanticAttributes.MESSAGING_CONSUMER_ID]: workerName,
+            [SemanticAttributes.MESSAGING_MESSAGE_ID]: job.id,
+            [SemanticAttributes.MESSAGING_OPERATION]: "receive",
+            [BullMQAttributes.JOB_NAME]: job.name,
+            [BullMQAttributes.JOB_ATTEMPTS]: job.attemptsMade,
+            [BullMQAttributes.JOB_TIMESTAMP]: job.timestamp,
+            [BullMQAttributes.JOB_DELAY]: job.delay,
+            [BullMQAttributes.JOB_REPEAT_KEY]: job.repeatJobKey,
+            ...Instrumentation.attrMap(BullMQAttributes.JOB_OPTS, job.opts),
+            [BullMQAttributes.QUEUE_NAME]: job.queueName,
+            [BullMQAttributes.WORKER_NAME]: workerName,
+            [BullMQAttributes.WORKER_CONCURRENCY]: this.opts?.concurrency,
+            [BullMQAttributes.WORKER_LOCK_DURATION]: this.opts?.lockDuration,
+            [BullMQAttributes.WORKER_LOCK_RENEW]: this.opts?.lockRenewTime,
+            [BullMQAttributes.WORKER_RATE_LIMIT_MAX]: this.opts?.limiter?.max,
+            [BullMQAttributes.WORKER_RATE_LIMIT_DURATION]:
+              this.opts?.limiter?.duration,
+            // Limit by group keys was removed in bullmq 3.x
+            [BullMQAttributes.WORKER_RATE_LIMIT_GROUP]: (
+              this.opts?.limiter as any
+            )?.groupKey,
+          }),
+          kind: SpanKind.CONSUMER,
+          links: Instrumentation.dropInvalidLinks([
+            {
+              context: trace.getSpanContext(producerContext),
+            },
+          ]),
+        });
 
-        return await context.with(messageContext, async () => {
+        const consumerContext = trace.setSpan(currentContext, span);
+
+        return await context.with(consumerContext, async () => {
           try {
             const result = await original.apply(this, [job, ...rest]);
             return result;
           } catch (e) {
             throw Instrumentation.setError(span, e as Error);
           } finally {
-            if (job.finishedOn)
-              span.setAttribute(
-                BullMQAttributes.JOB_FINISHED_TIMESTAMP,
-                job.finishedOn,
-              );
-            if (job.processedOn)
-              span.setAttribute(
-                BullMQAttributes.JOB_PROCESSED_TIMESTAMP,
-                job.processedOn,
-              );
-            if (job.failedReason)
-              span.setAttribute(
-                BullMQAttributes.JOB_FAILED_REASON,
-                job.failedReason,
-              );
+            span.setAttributes(
+              Instrumentation.dropInvalidAttributes({
+                [BullMQAttributes.JOB_FINISHED_TIMESTAMP]: job.finishedOn,
+                [BullMQAttributes.JOB_PROCESSED_TIMESTAMP]: job.processedOn,
+                [BullMQAttributes.JOB_FAILED_REASON]: job.failedReason,
```

Comment:
This wouldn't contain the error right?
I'm really interested in the format of that `JOB_FAILED_REASON` key too. Can I see this in the test app already? https://github.com/appsignal/test-setups/pull/211

Reply:
This contains the error message that caused the job to be re-attempted. Feel free to contribute a way to enqueue an erroring job with attributes in the test setup, but perhaps the easiest way to see it in action is to run the corresponding unit test with `printSpans` uncommented.

Reply:
Note that I didn't add this attribute, I just moved it around. It may get eliminated as part of a later cleanup of attributes that I intend to do. Marking this as resolved for now; unresolve if you disagree.